### PR TITLE
feat: user menu dropdown and click outside action

### PR DESCRIPTION
**Pull request by Michael-Liendo**

### CODE REVIEW

File: src/routes/(app)/components/Navbar/Profile.svelte [hunk 1]
```diff
@@ -20,5 +28,36 @@
...
+          <div class="font-medium truncate">{$page.data.user.email}!</div>
+        </div>
+        <ul class="py-2 text-sm text-gray-700 dark:text-gray-200">
+          <li>
+            <a
+              href="#"
+              class="block px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
+              >Settings</a
+            >
+          </li>
```
Comment (LeoBorai) [inline src/routes/(app)/components/Navbar/Profile.svelte]: I think we dont need this as of today, given that settings page doesnt exists

File: src/lib/actions/click-outside.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/** Dispatch event on click outside of node */
+export function clickOutside(node: any) {
+  const handleClick = (event: MouseEvent) => {
+    if (node && !node.contains(event.target) && !event.defaultPrevented) {
+      node.dispatchEvent(new CustomEvent('clickOutside', node));
```
Comment (LeoBorai) [inline src/lib/actions/click-outside.ts]: I think the `CustomEventInit` is not required.

```suggestion
      node.dispatchEvent(new CustomEvent('clickOutside'));
```

File: src/lib/actions/click-outside.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/** Dispatch event on click outside of node */
+export function clickOutside(node: any) {
+  const handleClick = (event: MouseEvent) => {
+    if (node && !node.contains(event.target) && !event.defaultPrevented) {
```
Comment (LeoBorai) [inline src/lib/actions/click-outside.ts]: Here we can assume this value is a `Node`.

```suggestion
    if (node && !node.contains(event.target as Node) && !event.defaultPrevented) {
```

File: src/lib/actions/click-outside.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/** Dispatch event on click outside of node */
+export function clickOutside(node: any) {
```
Comment (LeoBorai) [inline src/lib/actions/click-outside.ts]: `node` here is actually a `HTMLElement`

```suggestion
export function clickOutside(node: HTMLElement) {
```

**Review by LeoBorai — CHANGES_REQUESTED** [review]
Some help with types and remove not required nav item.
Remaining bits are looking great!

**Comment by Michael-Liendo** [comment]
Thanks for the help!

**Comment by LeoBorai** [comment]
> Thanks for the help!

Anytime!

**Review by LeoBorai — APPROVED** [review]

**Pull request merged**